File: articles/security-failures-as-algorithmic-failures/haskell/SecurityFailures.hs
module Main where

weights :: [Double]
weights = [0.08,0.10,0.10,0.10,0.08,0.08,0.08,0.08,0.08,0.08,0.06,0.05,0.03]

score :: Double
score = 100 * sum (map (*0.65) weights)

main :: IO ()
main = print score

File: articles/security-failures-as-algorithmic-failures/rust/security_failures.rs
fn main() {
    let weights = [0.08,0.10,0.10,0.10,0.08,0.08,0.08,0.08,0.08,0.08,0.06,0.05,0.03];
    let score: f64 = weights.iter().map(|w| w * 0.65).sum::<f64>() * 100.0;
    println!("{:.3}", score);
}
